HHS/Head-Start-TTADP
# [TTAHUB-2241] Update target populations

## Description of change

This adds, renames, and sorts the target populations on both the AR and TR.

Please see the JIRA ticket for a complete list of changes:
https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2241

In addition this fixes the JSON arrays in the 'EventReportPilots' table 'data' column to be correct arrays (with help from @GarrettEHill). 

## How to test

**Note:** You might need to run yarn deps to get latest common changes.

1. Review all changes.
2. Run the below queries before hand.
3. Run the migration.
4. Run the queries after (should be blank).
5. Create or edit a AR and TR change the reasons and target populations (verify updated and new values).
6. Verify TR shows the correct sort order (frequently used at the top).
7. Check the filters on all relevant pages make sure it reflects updates and filters.

**Queries to test:**

```
-- AR
SELECT id, "targetPopulations" FROM "ActivityReports" WHERE "targetPopulations" @> ARRAY['Preschool (ages 3-5)']::varchar[];
SELECT id, "targetPopulations" FROM "ActivityReports" WHERE "targetPopulations" @> ARRAY['Pregnant Women']::varchar[];
-- TR
SELECT data->>'targetPopulations' FROM "EventReportPilots" WHERE data->'targetPopulations' @> '"Preschool (ages 3-5)"';
SELECT data->>'targetPopulations' FROM "EventReportPilots" WHERE data->'targetPopulations' @> '"Pregnant Women/Pregnant People"';
```

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2241


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
